spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'POSTS', type: :request do
+  describe 'GET POSTS #index' do
+    before(:each) { get user_posts_path user_id: 20 }
+
+    it 'server return 200 then server is ok' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'template does rendering correctly' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'shows body content of index' do
+      expect(response.body).to include('<h1>show user posts details id</h1>')
+    end
+  end
+
+  describe 'GET POSTS #show' do
+    before(:each) { get user_posts_path user_id: 20, id: 30 }
+
+    it 'return 200' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'shows the correct show template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'shows body content of posts show' do
+      expect(response.body).to include('<h1>show user posts details</h1>')
+    end
```

Comment:
- [ ] This makes your test failing, kindly target the text that is included in the body instead of using ('<h1> show user posts details id</h1>') use ('show user posts details id') to target that text.